docs/vcl.md

```diff
@@ -0,0 +1,80 @@
+## Introduction and Overview
+
+VCL is a communication library in VPP that provides a socket-based interface for client/server communication with VPP. It is designed to enable client applications to establish connections with VPP, send and receive control messages, and perform configuration and management operations.
+
+## VCL feature in CalicoVPP
+
+### Enabling VCL feature in CalicoVPP
+
+To enable vcl in your calicoVPP cluster, make sure parameter is set here:
+
+```yaml
+# dedicated configmap for VPP settings
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: calico-vpp-config
+  namespace: calico-vpp-dataplane
+data:
+  CALICOVPP_FEATURE_GATES: |-
+  {
+    "vclEnabled": true
+  }
+```
+
+### Using VCL in pods:
+
+To enable VCL on a pod, use the following annotation:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: samplepod
+  annotations:
+    "cni.projectcalico.org/vppVcl": "enable"
+```
+This will trigger vpp to create the VCL socket and add a session, which is an abstraction that allows applications to send and receive data over the network.
+
+### Sockets
+
+VCL in VPP uses a socketfile: a Unix domain socket used for communication between endpoints. This allows sessions to initiate connections.
```

Comment:
The VPP Hoststack exposes an abstract socket, it's a unix socket over which applications communicate with VPP - leveraging the VCL - in order to open or accept L4+ connections (e.g. TCP) directly in VPP.